src/dotnet/ReSharperPlugin.FluentAssertions/QuickFixes/NUnitAssertMigrationQuickFix.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using JetBrains.Application.Progress;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.QuickFixes;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.TextControl;
+using JetBrains.Util;
+using ReSharperPlugin.FluentAssertions.Helpers.NUnit;
+using ReSharperPlugin.FluentAssertions.Highlightings;
+
+namespace ReSharperPlugin.FluentAssertions.QuickFixes
+{
+    /// <inheritdoc />
+    [QuickFix]
+    public class NUnitAssertMigrationQuickFix : QuickFixBase
+    {
+        private readonly NUnitAssertMigrationHighlighting _highlighting;
+
+        public NUnitAssertMigrationQuickFix(NUnitAssertMigrationHighlighting highlighting)
+        {
+            _highlighting = highlighting;
+        }
+
+        /// <inheritdoc />
+        protected override Action<ITextControl> ExecutePsiTransaction(ISolution solution, IProgressIndicator progress)
+        {
+            var invocationExpression = _highlighting.InvocationExpression;
+            var factory = CSharpElementFactory.GetInstance(invocationExpression);
+            var migrationServices = solution.GetComponents<INUnitAssertMigrationService>();
+
+            var migrationService = migrationServices.FirstOrDefault(x => x.CanMigrate(invocationExpression));
```

Comment:
Might be just personal preference, but I wouldn't make it so complicated by having an interface etc..

A better way might be having a base problem analyzer, base highlighting, and base quickfix. Most of the NUnit logic is then in the base class, and the derived class can make UI texts more specific, for instance like `Replace Assert.Null with obj.Should().BeNull()`

Reply:
`Assert.Null`, `Assert.NotNull`, `Assert.IsNotEmpty` - this simple invocations, they can implement base class, but NUnit Assert contains more complex methods for example: `Assert.That`, `Assert.Throws`
They contain much more complex logic.
Therefore, the interface suits me more, but maybe these are really personal preferences :-)
I also thought about more specific messages, I add this it is a little later

Reply:
I have a question. 
How I can create InvocationExpression for unit testing, without all testing infrastructure for integration testing?
If it is possible of course :-)

Reply:
I fixed your remark in commit https://github.com/matkoch/resharper-fluentassertions/pull/6/commits/cc0710f3739159b0d5f06343f9df808462e61df2